source/slang/slang-compiler-tu.cpp

```diff
@@ -0,0 +1,79 @@
+// slang-compiler-tu.cpp: Compiles translation units to target language
+// and emit precompiled blobs into IR
+
+#include "../core/slang-basic.h"
+#include "slang-compiler.h"
+#include "slang-ir-insts.h"
+
+namespace Slang
+{
+    SLANG_NO_THROW SlangResult SLANG_MCALL Module::precompileForTargets(
+        DiagnosticSink* sink,
+        EndToEndCompileRequest* endToEndReq,
+        TargetRequest* targetReq)
+    {
+        auto module = getIRModule();
+        Slang::Session* session = endToEndReq->getSession();
+        Slang::ASTBuilder* astBuilder = session->getGlobalASTBuilder();
+        Slang::Linkage* builtinLinkage = session->getBuiltinLinkage();
+        Slang::Linkage linkage(session, astBuilder, builtinLinkage);
```

Comment:
I am not very sure if this is correct. A linkage is supposed to be self-contained with no references to modules/IComponentTypes defined in another linkage. But the logic here is creating a new linkage, and use that linkage to create CompositeComponentType whose elements are in a different linkage. This may lead to weird referencing errors inside the compiler.

Reply:
Would it be sufficent to use endToEndReq()->getLinkage() instead of creating a new one?

Reply:
When I do that, I get this error:

```
$ ./build/Release/bin/slang-test.exe tests/library/p
Supported backends: fxc dxc glslang spirv-dis visualstudio genericcpp llvm spirv-opt
passed test: 'tests/library/precompiled-dxil.slang'
error: result code = -1
standard error = {
dxc 1.7: : error : error: invalid profile cs_5_1
}
standard output = {
}

FAILED test: 'tests/library/precompiled-dxil.slang.1'
passed test: 'tests/library/precompiled-dxil.slang.2'

===
66% of tests passed (2/3)
===

failing tests:
---
tests/library/precompiled-dxil.slang.1
---
```

Reply:
This error is due to the profile being incorrect. You need to make sure to create a temporary target request with the right profile, and generate code with that target request.

Reply:
Okay, will fix in next change.